CONTRIBUTING.md

```diff
@@ -22,47 +22,83 @@ _Notation:_ We prefix any command line input with "`>>>`" and generally show the
 
     You create this repository from the command line on your computer, using
-   ```bash
-   >>> git clone git@github.com:JoeCoolDummy/oomph-lib.git
-   ```
-   This command will create a new directory, `oomph-lib` which contains all the code and the relevant Git information. Have a look around:
-   ```bash
-   >>> cd oomph-lib
-   >>> ls -l
-   ```
-
-The official (`upstream`) repository has two key branches: `main` and `development`. These have now made it onto your computer. You can list the branches as follows:
+    ```bash
+    >>> git clone git@github.com:JoeCoolDummy/oomph-lib.git
+    ```
+    This command will create a new directory, `oomph-lib` which contains all the code and the relevant Git information.
+
+    If you haven't connected to GitHub via SSH before, you might encounter the following error:
+    ```bash
+    Cloning into 'oomph-lib'...
+    git@github.com: Permission denied (publickey).
+    fatal: Could not read from remote repository.
+
+    Please make sure you have the correct access rights
+    and the repository exists.
+    ```
+    To address this issue complete the following steps (links to the relevant GitHub resources have been provided):
+    1. [Generate a new SSH key if you don't have one.](https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#generating-a-new-ssh-key) 
+    2. [Add the new SSH key to the ssh-agent.](https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent) 
+    3. [Add the new SSH key to your GitHub account.](https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
+    
+    More details, if required, can be found on [Connecting to GitHub with SSH](https://docs.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh).
+    
+    
+    For consistency, we will use SSH commands in this document wherever possible.
+    If you would prefer to use the HTTPS-based equivalents instead, replace all occurrences of 
+    ```bash
+    git@github.com:JoeCoolDummy/oomph-lib.git
+    ``` 
+    with 
+    ```bash
+    https://github.com/JoeCoolDummy/oomph-lib
+    ```
+    **NOTE:** Prior to August 13, 2021, HTTPS-based commands required you to provide a username and a password. [Password-based authentication has now been deprecated by GitHub, and replaced by "token-based authentication"](https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/). Therefore, you will need to generate your own PAT. Instructions on how to do this can be found on [Creating a personal access token](https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token).
```

Comment:
I suggest we delete the comment about deprecation of password access. Water under the bridge